edx_sga/static/js/src/edx_sga.js

```diff
@@ -142,6 +142,15 @@ function StaffGradedAssignmentXBlock(runtime, element) {
                 });
             });
 
+            // Set up submission download links
+            var submissionDownloadedHtml = "<i class=\"fa fa-check-circle\" " +
+                "aria-hidden=\"true\" title=\"Submission downloaded\"></i>";
+            $("a.submission-filename").on("click", function () {
+                if (!$(this).prev("i").length) {
+                   $(this).before(submissionDownloadedHtml);
+                }
```

Comment:
This is hard to understand. Can you rewrite it so your intentions are easy to understand or add a comment which will explain it?

Reply:
To avoid doing a shamanistic ritual code, you can just add an extra class to the DOM element, like `.downloaded-indicator` or something like that and just toggle it (hide/show) when needed.